# HG changeset patch
# User John Donoghue <john.donoghue@ieee.org>
# Date 1381777074 14400
#      Mon Oct 14 14:57:54 2013 -0400
# Node ID 8680768efa8a211f791ec41f371e39f95365fdf9
# Parent  0c5f50706ba344387fb6b1fe789ae03f90ef175e
Enable copy of selected news browser text

* libgui/src/news-dock-widget.cc, libgui/src/news-dock-widget.h
 (news_doc_widget::copyClipboard): New function.

diff --git a/libgui/src/news-dock-widget.cc b/libgui/src/news-dock-widget.cc
--- a/libgui/src/news-dock-widget.cc
+++ b/libgui/src/news-dock-widget.cc
@@ -108,16 +108,25 @@ news_dock_widget::display_news (const QS
       if (news.contains ("critical-news-event") && ! isVisible ())
         setVisible (true);
     }
   else
     browser->setHtml (fixed_news);
 }
 
 void
+news_dock_widget::copyClipboard ()
+{
+  if (browser->hasFocus ())
+    {
+      browser->copy ();
+    }
+}
+
+void
 news_reader::process (void)
 {
   // Run this part in a separate thread so Octave can continue to run
   // while we wait for the page to load.  Then emit the signal to
   // display it when we have the page contents.
 
   QString url = base_url + "/" + page;
   std::ostringstream buf;
diff --git a/libgui/src/news-dock-widget.h b/libgui/src/news-dock-widget.h
--- a/libgui/src/news-dock-widget.h
+++ b/libgui/src/news-dock-widget.h
@@ -36,16 +36,19 @@ public:
   news_dock_widget (QWidget *parent = 0);
 
   void load_news (void);
 
 protected slots:
 
   void display_news (const QString& news);
 
+  /* from octave_dock_widget */
+  void copyClipboard ();
+
 private:
 
   QTextBrowser *browser;
 };
 
 class news_reader : public QObject
 {
   Q_OBJECT
